mod/friends/start.php

```diff
@@ -231,10 +225,68 @@ function _elgg_send_friend_notification($event, $type, $object) {
 		'object' => $user_one,
 		'friend' => $user_two,
 	];
-	
+
 	return notify_user($user_two->guid, $object->guid_one, $subject, $body, $params);
 }
 
-return function(\Elgg\EventsService $events, \Elgg\HooksRegistrationService $hooks) {
-	$events->registerHandler('init', 'system', '_elgg_friends_init');
-};
+/**
+ * Add ACCESS_FRIENDS to the available access levels
+ *
+ * @param string $hook         "access:collections:write"
+ * @param string $type         "user"
+ * @param array  $access_array Access array
+ * @param array  $params       Hook params
+ *
+ * @return array
+ */
+function _elgg_friends_write_access($hook, $type, $access_array, $params) {
+
+	// rebuild array, putting friends 1st or 2nd
```

Comment:
is this really necessary? MySQL will optimize query to run the fastest am i right? Did you run tests to confirm this reorder is needed?